DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/buffered_host_streaming/src/streaming_with_api.hpp

```diff
@@ -0,0 +1,271 @@
+#ifndef __STREAMING_WITH_API_HPP__
+#define __STREAMING_WITH_API_HPP__
+
+#include <algorithm>
+#include <numeric>
+#include <queue>
+#include <thread>
+#include <vector>
+
+#include <CL/sycl.hpp>
+#include <CL/sycl/INTEL/fpga_extensions.hpp>
+
+#include "common.hpp"
+#include "HostStreamer.hpp"
+
+using namespace sycl;
+using namespace std::chrono;
+
+////////////////////////////////////////////////////////////////////////////////
+// forward declare functions

```

Comment:
Not clear why you declare all your functions.
Is this necessary?
This makes the tutorial more complex and less Python-esque

Reply:
This was an internal suggestion. People wanted the main function at the top so you don't have to scroll past a bunch of functions to see it. I could put them in a header file, but that would be another file and we kind of want to reduce the number of files.

> This makes the tutorial more complex and less Python-esque

I'd argue this is okay, since it is not Python ;)